### DISCUSSION TITLE
안녕하세요! Dependency 관련 질문드립니다!

### DISCUSSION THREAD

**Discussion by Hansolkkim** [nsoojin/MiniSuperApp-fastcampus#18] [아무 질문이나 환영합니다]
현재 2부 수강중인데, 모듈화 관련해서 의문점이 생겨 질문드립니다.

강의 2부 4번째 강의에서, AppHome을 모듈 분리하기 전에 TransportHome을 모듈 분리하셨었는데 이 때는 앱 실행이 잘 되고, AppHome을 모듈 분리한 후에 앱 실행이 안되어 Transport 패키지에 ModernRIBs, Finance, Topup과 관련된 Dependency를 추가하셨었습니다.

혹시, Transport 패키지에 ModernRIBs, Finance, Topup Dependency를 추가하지 않았음에도 앱은 정상적으로 실행됐던 이유에 대해 알 수 있을까요?
(그리고 현재 확인해보니, 강의 중에 Transport 패키지에 CombineUtil 라이브러리와 관련된 Platform 패키지에 대한 Dependency를 추가하지 않았음에도 `import CombineUtil` 코드는 정상적으로 실행이 됐었습니다!)

---
괄호로 붙인 상황에 대해 제가 생각한 원인은, TransportHome 라이브러리에서 FinanceRepository, Topup 라이브러리에 대한 Dependency를 추가했고, TransportHome을 빌드하기 전에 해당 라이브러리들을 빌드하면서 해당 라이브러리가 CombineUtil에 대한 Dependency를 이미 가지고 있기 때문에 TransportHome이 빌드될 시점에는 이미 해당 Dependency를 가지고 있게 되기 떄문이라고 생각했습니다.
혹시 이런 원인이 맞을까요?

**Comment by nsoojin** [comment]
안녕하세요.
네 저도 Hansolkkim님이 추측하신 것이 원인이라고 생각합니다. 빌드 로그를 보고 무엇이 먼저 빌드 되었는지 확인해보시면 좋을거 같습니다. 추측하신대로 TransportHome보다 CombineUtil을 가지고 있던 다른 패키지가 먼저 빌드되었다면 맞을테고요. 

그리고 하나 더 확인해보실 수 있는건 TransportHome 패키지만 빌드하는 스킴을 만들어서 빌드해보시는 겁니다. 그러면 디펜던시가 빠진게 있다면 빌드가 실패할 것입니다.
